### DISCUSSION TITLE
what is no-else-return supposed to do?

### DISCUSSION THREAD

**Discussion by JonFranchi** [eslint/eslint#14097] [Q&A]
I have some issues to clean up which showed up in the latest linter upgrade, but I can't tell if the results of --fix are buggy or intentional. It doesn't seem to match the description in the linter rule, but I can't find anybody else talking about it.

ex. 1:
```
  if (conditionalCheck)) {
    return 'value';
  } else if (otherConditionalCheck) {
    return 'otherValue';
  }
```
Is fixed as:

```
 if (conditionalCheck)) {
    return 'value';
  } if (otherConditionalCheck) {
    return 'otherValue';
  }
```

ex. 2:
```
  if (conditionalCheck)) {
    return { status: resultType.invalid };
  } else if (otherConditionalCheck) {
    someObject.val = otherObject.otherVal
  }
... do some other stuff
    return { status: resultType.success };
```
Is fixed as:

```
  if (conditionalCheck)) {
    return { status: resultType.invalid };
  } if (otherConditionalCheck) {
    someObject.val = otherObject.otherVal
  }
... do some other stuff
    return { status: resultType.success };
```

In the first one, why is it leaving the `if` block on the same line the other one ends?
In the second one, same question, but this makes me question the definition of the rule which says:

> `If an if block contains a return statement, the else block becomes unnecessary. Its contents can be placed outside of the block.`

That's not actually true here.  We're short circuiting, then ensuring on not-guaranteed-thing is set, then proceeding with the code.  The second conditional definitely cannot be placed outside the block.  Though I agree it doesn't need to be chained to the previous conditional and works perfectly fine as it's own if block.

Any thoughts on this?  The biggest thing is the fix leaving the conditional starting on the same line the previous one ends.  That can't be deliberate, right?

**Comment by tiagoajacobi** [comment]
I'm having the same issue here, decided to start using more eslint auto-fix, and just saw that issue today.
